internal/target/repository_address.go

```diff
@@ -0,0 +1,20 @@
+package target
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+func fetchAddress(ctx context.Context, r db.Reader, targetId string) (*Address, error) {
+	const op = "target.fetchAddress"
+	var address *Address
+	if err := r.SearchWhere(ctx, &address, "target_id = ?", []any{targetId}); err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	if address.TargetAddress == nil {
+		return nil, nil
```

Comment:
I know this pattern is pervasive in our code, but I think it's demonstrably dangerous (message me for details), and I would much sooner we return an error in these cases. What do you think?